plugins/jobs/web_client/views/JobListWidget.js

```diff
@@ -24,21 +24,28 @@ var JobListWidget = View.extend({
 
     initialize: function (settings) {
         var currentUser = getCurrentUser();
+        this.showAllJobs = !!settings.allJobsMode;
         this.columns = settings.columns || this.columnEnum.COLUMN_ALL;
         this.filter = settings.filter || {
-            userId: currentUser.id
+            userId: currentUser ? currentUser.id : null
         };
         this.typeFilter = {};
         this.statusFilter = {};
 
         this.collection = new JobCollection();
+        if (this.showAllJobs) {
+            this.collection.resourceName = 'job/all';
+        }
         this.collection.sortField = settings.sortField || 'created';
         this.collection.sortDir = settings.sortDir || SORT_DESC;
         this.collection.pageLimit = settings.pageLimit || this.collection.pageLimit;
 
         this.collection.on('g:changed', function () {
+            this.typeFilter = {};
+            this.statusFilter = {};
```

Comment:
Hi @manthey, I see what you mean with the behavior. I made this change is because I thought I was experiencing missing records (but it was actually filtered out). Maybe it is the filter checkbox needs to be fixed instead.